web/src/unread_ops.js

```diff
@@ -206,6 +206,42 @@ export function mark_as_unread_from_here(
         narrow = JSON.stringify(message_lists.current.data.filter.terms());
     }
     message_lists.current.prevent_reading();
+
+    // If we have already fully fetched the current view, we can
+    // send the server the set of IDs to update, rather than
+    // updating on the basis of the narrow.
+    let message_ids_to_update;
+    if (message_lists.current.data.fetch_status.has_found_newest()) {
+        message_ids_to_update = message_lists.current
+            .all_messages()
+            .filter(
+                (msg) =>
+                    (include_anchor && msg.id >= message_id) ||
+                    (!include_anchor && msg.id > message_id),
+            )
+            .map((msg) => msg.id);
+    }
+
+    if ((message_ids_to_update?.length ?? 0) < 200) {
```

Comment:
I think this might handle the case where we don't have the view fully fetched incorrectly? We want to ask the server to do it correctly for us in that case, but you're doing the opposite.

Reply:
Yeah, Idk how i missed something like this. I'll change it right away.